gh-actions/jq/jq.ts

```diff
@@ -17,20 +18,22 @@ const run = async (): Promise<void> => {
     core.info(`options: ${options}`)
     core.info(`filter: ${filter}`)
 
-    const args: string[] = []
-    args.push(options)
-    args.push(`${filter}`)
-
-    console.log(`jq ${args.join(' ')}`)
-    console.log(`> '${input}'`)
-    const proc = spawn('jq', args)
-    proc.process.stdin.setEncoding('utf-8')
-    proc.process.stdin.write(`${input}`)
-    proc.process.stdin.end()
+    // preferably use spawn/stdin
+    const shellCommand = `echo '${input}' | jq ${options} '${filter}'`
+    console.log(`Running shell command: ${shellCommand}`)
+    const proc = spawn('sh', ['-c', shellCommand])
     const response = await proc
     const stdout = response.stdout
     core.setOutput('value', stdout)
+    if (envVar) {
+      process.env[envVar] = stdout
+      core.exportVariable(envVar, stdout)
+    }
   } catch (error) {
+    const e = error as Record<'stderr', string>
+    if (e.stderr) {
+      console.error(`jq stderr: ${e.stderr}`)
+    }
```

Comment:
Can it have stderr *and* be an instance of Error with message?

If not, I'd suggest doing the Error check first, and the stderr check in an 'else'.

Reply:
i wasnt sure about that so left it separate